### PR TITLE
Added lower boundaries to email triggers and email details

### PR DESCRIPTION
**Pull request by MurasatoNaoya**
The plants at the LMNH are sensitive to both higher and lower temperatures and soil moisture, this pull request accounts for that lower bound, while also update slightly the the emails themselves, specifically mentioning that a given value is too low or too high.

### CODE REVIEW

File: pipeline/emailing.py [hunk 1]
```diff
@@ -10,34 +10,52 @@
 from load import create_connection
 
 
-def send_email(plant_id: int, value: float, value_type: str) -> None:
-    """Sends an email to the correct botanist if the plants conditions are worrying."""
+def send_email(plant_id: int, value: float, value_type: str, condition:str ) -> None:
+    """
+    Sends an email to the correct botanist if the plants conditions are worrying.
+    Example inputs - plant_id: 1, value: 10000, value_type: 'temperature', condition: 'exceeded'.
```
Comment (CalebMathers) [inline pipeline/emailing.py:16]: Nice to have example inputs in docstring

File: pipeline/emailing.py [hunk 2]
```diff
@@ -10,34 +10,52 @@
...
+    """
+
     load_dotenv()
-    html = generate_html(plant_id, value, value_type)
+    html = generate_html(plant_id, value, value_type, condition)
     client = boto3.client("ses", region_name="eu-west-2")
     message = MIMEMultipart()
-    message["Subject"] = "Plant ALERT!"
+    message["Subject"] = f"PLANT ALERT! ACCEPTABLE {value_type.upper()} LEVEL {condition.upper()}"
```
Comment (CalebMathers) [inline pipeline/emailing.py:24]: Good subject

File: pipeline/emailing.py [hunk 3]
```diff
@@ -10,34 +10,52 @@
...
+    #         ENV['TO_EMAIL4']
+    #     ],
+    #     RawMessage={
+    #         'Data': message.as_string()
+    #     }
+    # )
+
     client.send_raw_email(
-        Source=ENV['FROM_EMAIL'],
+        Source='trainee.andrew.mcwilliam@sigmalabs.co.uk',
```
Comment (CalebMathers) [inline pipeline/emailing.py:44]: Could change this back to get it from the ENV when merging to main

Comment (megansam5) [reply pipeline/emailing.py:44]: just before we submit this, ill change it to the botanist email, so we can send it directly to them

File: pipeline/emailing.py [hunk 4]
```diff
@@ -77,11 +108,12 @@ def generate_html(plant_id: int, value: float, value_type: str) -> str:
     </head>
     <body>
         <div class="container">
-            <h2>Automated Botany Alert</h2>
+            <h2>Automated Botany Alert: Acceptable {capitalised_value_type} Range {condition.capitalize()}</h2>
             <p>Dear {botanist_name} ({botanist_email}),</p>
             <p>
                 This is an automated email to inform you that the {value_type} for the plant
-                <strong>{plant_name}</strong> with the ID <strong>{plant_id}</strong> has recorded a value of <strong>{value}{unit}</strong>.
+                <strong>{plant_name}</strong> with the ID <strong>{plant_id}</strong> has recorded a value of <strong>{value}{unit}</strong>,
```
Comment (megansam5) [inline pipeline/emailing.py:115]: nice ability to alter to the condition

File: pipeline/transform.py [hunk 5]
```diff
@@ -24,9 +24,16 @@ def transform(needs_a_clean: pd.DataFrame):
...
+
+        if row['temperature'] <= 5:
+            send_email(row['plant_id'], row['temperature'], 'temperature', 'not met')
+        
+        if row['soil_moisture'] >= 90:
+            send_email(row['plant_id'], row['soil_moisture'], 'soil moisture', 'exceeded')
+
         if row['soil_moisture'] < 30:
-            send_email(row['plant_id'], row['soil_moisture'], 'soil moisture')
+            send_email(row['plant_id'], row['soil_moisture'], 'soil moisture', 'not met')
```
Comment (megansam5) [inline pipeline/transform.py:36]: again, good ability to change depending on condition.

File: pipeline/emailing.py [hunk 6]
```diff
@@ -10,18 +10,25 @@
...
-    message["Subject"] = f"Plant {plant_id} Alert!"
+    
+    message["Subject"] = f"PLANT ALERT! ACCEPTABLE {value_type.upper()} LEVEL {condition.upper()}"
     body = MIMEText(
         html,
         "html")
     message.attach(body)
 
+    # The addresses the emails will be sent, 
+    # should be specified in your local env variables.
```
Comment (MurasatoNaoya) [inline pipeline/emailing.py:31]: Updated where the emails will be sent, not to my test email, but to the emails stored locally in an .env file.

File: pipeline/emailing.py (hunk 2 above)
Comment (Ruyzambrano) [reply pipeline/emailing.py:24]: Surely you mean unacceptable?

Comment (MurasatoNaoya) [reply pipeline/emailing.py:24]: The email title would look like "PLANT ALERT! ACCEPTABLE (TEMPERATURE / SOIL MOISTURE) EXCEEDED / NOT MET".  Unacceptable temperature or unacceptable soil moisture seemed too general.

**Pull request merged**